models/oci/oci.go

```diff
@@ -112,3 +119,28 @@ func getLayerMediaType(extension string) types.MediaType {
 	}
 	return types.MediaType(fmt.Sprintf("%s.%s", oci.CanonicalMediaTypePrefix, extension))
 }
+
+// function to pull models from the public oci repository
+func PullFromOCIRegistry(dirPath ,registryURL, imageTag string) error {
+	// Create a new file store
+	fs, err := file.New(dirPath)
+	if err != nil {
+		return ErrFileNotFound(err)
+	}
+
+	defer fs.Close()
+	ctx := context.Background()
+
+	// Connect to remote registry
+	repo, err := remote.NewRepository(registryURL)
+	if err != nil {
+		return ErrConnectingToRegistry(err)
+	}
+
+	_ , err = oras.Copy(ctx, repo, imageTag, fs, imageTag, oras.DefaultCopyOptions)
+	if err != nil {
+		return ErrGettingImage(err)
```

Comment:
The suggested remediation and probable cause for this error code are pretty terribly written.

https://github.com/meshery/meshkit/blob/75d94b9fe6be1111aef6f9942bca03db9602956d/models/oci/error.go#L44